src/explore-education-statistics-common/src/components/Pagination.tsx

```diff
@@ -0,0 +1,122 @@
+import { ArrowLeft, ArrowRight } from '@common/components/ArrowIcons';
+import generatePageNumbers from '@common/components/util/generatePageNumbers';
+import { useDesktopMedia } from '@common/hooks/useMedia';
+import classNames from 'classnames';
+import React, { ReactNode } from 'react';
+
+interface Props {
+  currentPage: number;
+  nextPrevLinkRenderer: ({
+    children,
+    className,
+    pageNumber,
+    rel,
+  }: {
+    children: ReactNode;
+    className: string;
+    pageNumber: number;
+    rel: 'next' | 'prev';
+  }) => ReactNode;
```

Comment:
Do we need to separate a separate prop for this? Would suggest that we consolidate both of these into a single prop:

```ts
interface LinkRenderProps {
  'aria-current'?: 'page' | undefined;
  'aria-label'?: string;
  children: ReactNode;
  className: string;
  pageNumber: number;
  rel?: 'next' | 'prev';
}

interface PaginationProps {
  renderLink: (props: LinkRenderProps) => ReactNode;
} 
```

Note that aria props are using snake-case. The intention of this would be so that we can just spread these props straight into whatever link component we're using:

```tsx
<Pagination
  renderLink={{ pageNumber, ...props }) => 
    <Link {...props} to={`?page=${pageNumber}`} />
  }
/>
```

Think we could also go a little further and return the link's `href` (or `to` to better suit our Link components) as part of the render props too. We'd want to set the base URL with something like:
```ts
interface PaginationProps {
  baseUrl?: string;
}
```

The `href` / `to` render prop would then return a URL that concatenates the `baseUrl` if it has been set, or returns the query string like `?page=${pageNumber}` if it hasn't. Might need the `qs` package to help with parsing query parameters, but probably possible to do in a simple manner without.

Note that the `baseUrl` could include its own query string, so we should bear that in mind when concatenating the query string to it.

We should then be able to simplify our render prop to something like:
```tsx
<Pagination
  renderLink={props => <Link {...props} />}
/>
```